File: back/organisme-securite-sociale/src/main/java/com/securitesociale/entity/Personne.java
package com.securitesociale.entity;

import com.securitesociale.entity.enums.Genre;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "personnes")
@Inheritance(strategy = InheritanceType.JOINED)
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class Personne {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(length = 100)
    private String nom;

    @Column(length = 100)
    private String prenom;

    @Enumerated(EnumType.STRING)
    private Genre genre;

    private LocalDate dateNaissance;

    @Column(length = 255)
    private String adresse;

    @Column(length = 15)
    private String telephone;

    @Column(unique = true)
    private String email;

    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "photo_id")
    private Media photo;

    @Column(name = "date_creation", nullable = false, updatable = false)
    private LocalDateTime dateCreation;

    @PrePersist
    protected void onCreate() {
        this.dateCreation = LocalDateTime.now();
    }
}

File: front/securite-sociale-front/src/types/personne.ts
// src/types/personne.ts
export const Genre = {
  M: 'M',
  F: 'F',
} as const;

export type Genre = (typeof Genre)[keyof typeof Genre];


export interface Personne {
  id: number;
  nom: string;
  prenom: string;
  dateNaissance: string;
  genre: Genre;
  adresse: string;
  telephone: string;
  email: string;
  photoId?: number;
  dateCreation: string;
}

export interface PersonneDTO {
  id?: number;
  nom: string;
  prenom: string;
  dateNaissance: string;
  genre: Genre;
  adresse: string;
  telephone: string;
  email: string;
  photoId?: number;
  dateCreation?: string;
}

export interface PersonneFormData {
  nom: string;
  prenom: string;
  dateNaissance: string;
  genre: Genre;
  adresse: string;
  telephone: string;
  email: string;
}

export interface PersonneSearchParams {
  nom?: string;
  email?: string;
  telephone?: string;
}